runtime/tests/checker/composite_type_fields_test.go

```diff
@@ -0,0 +1,231 @@
+	"github.com/stretchr/testify/require"
+)
+
+func TestCompositeTypeFields(t *testing.T) {
+	t.Parallel()
+
+	cases := map[string]struct {
+		code   string
+		result bool
+	}{
+
+		"int is a storable field": {`
+		contract Controller {
+
+			var n: Int
+
+			init(){
+				self.n = 1
+			}
+		}
+			`,
+			true,
+		},
+
+		"function is not a storable field": {`
+		contract Controller {
+
+			var fn: (():Int)
+
+			init(){
+				self.fn = fun(): Int {
+					return 1
+				};
+			}
+
+			pub fun test(): Int {
+					return 2
+			}
+		}
+			`,
+			false,
+		},
+
+		"[Int] is a storable field": {`
+		contract Controller {
+
+			var xs: [Int]
+
+			init(){
+				self.xs = [1, 2, 3]
+			}
+		}
+			`,
+			true,
+		},
+
+		"{Int: String} is a storable field": {`
+		contract Controller {
+
+			var m: {Int: String}
+
+			init(){
+				self.m = {}
+			}
+		}
+			`,
+			true,
+		},
+
+		"PublicAccount is a not storable field": {`
+		struct S {
+			var p: PublicAccount
+
+		}
+			`,
+			false,
+		},
+
+		"{Int: function} is a storable field": {`
+		contract Controller {
+
+			var m: {Int: ((): Int)}
+
+			init(){
+				self.m = {
+					1: fun(): Int {
+						return 1
+					}
+				}
+			}
+		}
+			`,
+			false,
+		},
+
+		"[function] is not a storable field": {`
+		contract Controller {
+
+			var operators: [(():Int)]
+
+			init(){
+				self.operators = []
+			}
+		}
+			`,
+			false,
+		},
+
+		"function field for struct is not storable": {`
+		struct MyStruct {
+			pub var fn: (():Int)
+
+			init() {
+				self.fn = fun(): Int {
+					return 1
+				};
+			};
+		}
+			`,
+			false,
+		},
+
+		"path field is not storable": {`
+		struct MyStruct {
+			pub var fn: (():Int)
+
+			init() {
+				self.fn = fun(): Int {
+					return 1
+				}
+			}
+		}
+			`,
+			false,
+		},
+
+		"nested field for resource is not storable": {`
+		contract S {
+			let r : @R
+
+			resource R {
+				// function field in nested composite type is not allowed
+				pub var fn: (():Int)
+
+				init() {
+					self.fn = fun(): Int {
+						return 1
+					}
+				}
+			}
+		}
+			`,
+			false,
+		},
+
+		"resource interface is storable if all fields are storable": {`
+		resource interface RI {
+			var r: Int
+			var s: String
+		}
+			`,
+			true,
+		},
+
+		"resource interface is not storable if one field is not storable": {`
+		resource interface RI {
+			var r: Int
+			var p: PublicAccount // PublicAccount is not a storable field
+		}
+
+		resource R {
+			var m : @{String: {RI}}
+		}
+			`,
+			false,
+		},
+	}
+
+	for caseName, testcase := range cases {
+		t.Run(caseName, func(t *testing.T) {
+			_, err := ParseAndCheck(t, testcase.code)
+
+			errmsg := fmt.Sprintf("failed test case: %v\n", testcase.code)
+			if testcase.result {
+				if err != nil {
+					cmd.PrettyPrintError(err, "", map[string]string{"": testcase.code})
+				}
+				// print the failed the cadence code if test case was broken
+				require.NoError(t, err, errmsg)
+			} else {
+				require.Error(t, err, errmsg)
```

Comment:
Check the actual errors here, not just that something went wrong: The checker could reject the tests for other reasons.

The common pattern is:

```
		errs := ExpectCheckerErrors(t, err, 1)

		assert.IsType(t, &sema.FieldTypeNotStorableError{}, errs[0])
```

`ExpectCheckerErrors` ensures the error is an error and has the correct number of child errors